core/deployment/src/main/java/io/quarkus/deployment/steps/ClassPathSystemPropBuildStep.java

```diff
@@ -17,15 +16,6 @@
 
 public class ClassPathSystemPropBuildStep {
 
-    @BuildStep
-    public void produce(BuildProducer<SetClassPathSystemPropBuildItem> producer, CurateOutcomeBuildItem curateOutcome) {
-        boolean truffleUsed = curateOutcome.getApplicationModel().getDependencies().stream()
-                .anyMatch(d -> d.getGroupId().equals("org.graalvm.polyglot"));
-        if (truffleUsed) {
-            producer.produce(new SetClassPathSystemPropBuildItem());
```

Comment:
Should we deprecate `SetClassPathSystemPropBuildItem` given it's a bit buggy and we probably don't want people to use it?

Reply:
I was on the fence, but as you bring it up, yeah, let's do it

Reply:
Done